frontend/packages/kubevirt-plugin/src/components/modals/cpu-pinning-modal/cpu-pinning-modal.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { Checkbox, Modal, Title } from '@patternfly/react-core';
+import { k8sPatch } from '@console/internal/module/k8s';
+import { withHandlePromise, HandlePromiseProps } from '@console/internal/components/utils';
+import { ModalComponentProps } from '@console/internal/components/factory';
+import { ModalFooter } from '../modal/modal-footer';
+import { getVMLikeModel } from '../../../selectors/vm';
+import { getCPUPinningPatch } from '../../../k8s/patches/vm/vm-cpu-patches';
+import { VMLikeEntityKind } from '../../../types';
+import './cpu-pinning-modal.scss';
+
+export const CpuPinningModal = withHandlePromise(
+  ({
+    vmLikeEntity,
+    isCPUPinned,
+    isOpen,
+    setOpen,
+    title = 'Devices & Resources',
+    handlePromise,
+    inProgress,
+    errorMessage,
+  }: CpuPinningModalProps) => {
+    const [isPinned, setIsPinned] = React.useState<boolean>(isCPUPinned);
+    const [showPatchError, setPatchError] = React.useState<boolean>(false);
+
+    const submit = async () => {
+      // eslint-disable-next-line promise/catch-or-return
+      if (isPinned !== isCPUPinned) {
+        handlePromise(
+          k8sPatch(
+            getVMLikeModel(vmLikeEntity),
+            vmLikeEntity,
+            await getCPUPinningPatch(vmLikeEntity, isPinned),
+          ),
+        )
+          .then(() => setOpen(false))
+          .catch(() => setPatchError(true));
+      } else {
+        setOpen(false);
+      }
+    };
+
+    const footer = (
+      <ModalFooter
+        errorMessage={showPatchError && errorMessage}
+        inProgress={inProgress}
+        onSubmit={submit}
+        onCancel={() => setOpen(false)}
+        submitButtonText="Save"
+      />
+    );
+
+    return (
+      <Modal
+        title={title}
+        isOpen={isOpen}
+        isSmall
+        onClose={() => setOpen(false)}
+        footer={footer}
```

Comment:
looks like the alignment of the footer is a bit off. I'm seeing a wrapping div around the footer which shouldn't be needed.
Do we need `footer={footer}`?

Reply:
yes, custom actions and loading animations. ill fix the left padding